### PR TITLE
fix(plugin): pass solution directory to plugin providers for path res…

### LINKED ISSUE #360: bug(engine): plugin providers don't receive SolutionDirectory for path resolution
## Summary

When running `scafctl run resolver` from the repo root with a solution auto-discovered at `./scafctl/solution.yaml`, plugin providers (e.g., `directory`) resolve relative paths from CWD instead of the solution file's directory. Builtin providers work correctly because they share the host context.

## Root Cause

The resolver phase intentionally does NOT set `WorkingDirectory` on the context (to avoid conflicts with `SolutionDirectory`). The `AbsFromContext` resolution order is:

1. `WorkingDirectory` (empty for resolvers -- by design)
2. `SolutionDirectory` (set correctly on host context)
3. `filepath.Abs(path)` (CWD fallback)

Builtin providers execute in-process and see `SolutionDirectory` directly from the context. However, **plugin providers run in a separate process** and receive context via the gRPC `ExecuteProviderRequest` proto. The `SolutionDirectory` context value is **not transmitted** in the proto message, so plugin providers always hit the CWD fallback.

## Reproduction

1. Solution at `./scafctl/solution.yaml`:
```yaml
spec:
  resolvers:
    sourceFiles:
      resolve:
        with:
          - provider: directory
            inputs:
              path: ./static
              operation: list
```
2. Files at `./scafctl/static/`
3. Run from repo root: `scafctl run resolver`
4. Error:
```
resolver execution failed: phase 1 failed: resolver "sourceFiles" failed:
provider execution failed: directory: directory does not exist:
/path/to/repo/static
```
Expected: resolves `./static` relative to `./scafctl/` (the solution directory).

**Workaround**: `scafctl run resolver --base-dir scafctl` or `cd scafctl` before running.

## Affected Area

- `pkg/plugin/grpc.go` -- `buildExecuteProviderRequest` does not include `SolutionDirectory`
- `pkg/provider/path.go` -- `AbsFromContext` correctly checks `SolutionDirectory` but the plugin never receives it
- Any official/plugin provider using relative paths (`directory`, `file`, etc.)

## Proposed Fix

In `buildExecuteProviderRequest` (or the plugin wrapper's `Execute` method), when `WorkingDirectory` is empty and `SolutionDirectory` is set on the context, include `SolutionDirectory` as the `working_directory` field in the proto request. This way plugin providers resolve paths the same way builtin providers do.

Alternative: add a dedicated `solution_directory` field to the proto and have the plugin SDK's `applyRequestContext` set it on the context so `AbsFromContext` works identically in both modes.

## Risks

- Fixing this could be a breaking change for users who accidentally relied on CWD behavior for plugin providers
- Catalog/bundle execution paths should be verified to remain unaffected (bundles already set `WorkingDirectory` via `bundleDir`)

### PR DESCRIPTION
**Pull request by abaker-9**
…olution

- fall back to SolutionDirectory when WorkingDirectory is not set in buildExecuteProviderRequest, matching AbsFromContext behavior
- add tests for fallback, precedence, and empty-context cases

Closes #360

### CODE REVIEW

File: pkg/plugin/grpc.go [hunk 1]
```diff
@@ -739,6 +739,11 @@ func buildExecuteProviderRequest(ctx context.Context, providerName string, input
 	// Working directory
 	if dir, ok := provider.WorkingDirectoryFromContext(ctx); ok {
 		req.WorkingDirectory = dir
+	} else if solDir, ok := provider.SolutionDirectoryFromContext(ctx); ok && solDir != "" {
+		// When no explicit working directory is set (e.g., resolver phase),
+		// fall back to the solution directory so plugin providers resolve
+		// relative paths the same way builtin providers do.
+		req.WorkingDirectory = solDir
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/plugin/grpc.go:746]: Fixed -- added dir != "" guard to match AbsFromContext behavior. Thanks!

[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/plugin/grpc.go:746]: Fixed -- added dir != "" guard to match AbsFromContext behavior. Thanks!

File: pkg/plugin/grpc_new_test.go [hunk 2]
```diff
@@ -173,6 +173,35 @@ func TestGRPCServer_ExecuteProvider_ExtendedContext(t *testing.T) {
...
+	})
+
+	t.Run("no directory set leaves working directory empty", func(t *testing.T) {
+		ctx := context.Background()
+
+		req, err := buildExecuteProviderRequest(ctx, "test-provider", []byte(`{}`))
+		require.NoError(t, err)
+		assert.Empty(t, req.WorkingDirectory)
+	})
+}
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply pkg/plugin/grpc_new_test.go:213]: Good catch, added a test case for WithWorkingDirectory(ctx, "") falling through to solution directory.
